libraries/standard/mqtt/src/mqtt.c

```diff
@@ -865,9 +874,12 @@ static MQTTStatus_t handlePublishAcks( MQTTContext_t * pContext,
 
     if( status == MQTTSuccess )
     {
+        pDeserialized->packetIdentifier = packetIdentifier;
+        pDeserialized->status = status;
+
         /* Invoke application callback to hand the buffer over to application
          * before sending acks. */
-        appCallback( pContext, pIncomingPacket, packetIdentifier, NULL );
+        appCallback( pContext, pIncomingPacket, pDeserialized );
```

Comment:
If the app callback is invoked within this function, the `MQTTDeserializedInfo_t * pDeserialized` output parameter shouldn't be required

Reply:
Yes, it's not required, the reason I added it was that this function is called by `handleIncomingAck`, which already declared an `MQTTDeserializedInfo_t` on the stack. Even though the struct shouldn't be very large, I didn't think it was necessary to declare another one in this function, and just passed the first one in via pointer.

Reply:
I don't think we should have the API be sending such a parameter. It seems like the common function `handleIncomingAcks` should be broken for handling the Publish and PinResponse/Subscribe Acks separately

Reply:
Removed the pointer parameter